### PR TITLE
OCPBUGS-35587: T-GM - ts2phc process restart doesn't update PTP syncState when the process recovered

### PR DESCRIPTION
**Pull request by aneeshkp**
ts2phc process termination only triggers FREERUN and doesn't recover back .
This PR fixes this issue by reading process down status and handle ts2phc event

### CODE REVIEW

File: plugins/ptp_operator/metrics/metrics.go [hunk 1]
```diff
@@ -131,6 +131,7 @@ func (p *PTPEventManager) ExtractMetrics(msg string) {
 	//  check if ptp4l or phc2sys process are dead
 	// if process is down fire an event
 	if strings.Contains(output, ptpProcessStatusIdentifier) {
+		log.Infof("output %s", output)
```
Comment (jzding) [inline plugins/ptp_operator/metrics/metrics.go:134]: is this temporary? too much logs.

File: plugins/ptp_operator/stats/stats.go [hunk 2]
```diff
@@ -236,7 +236,7 @@ func (s *Stats) GetStateState(processName string, iface *string) (ptp.SyncState,
 				return d[0].State, nil
 			}
 			for _, state := range d {
-				if *state.IFace == *iface {
+				if iface != nil && *state.IFace == *iface {
```
Comment (jzding) [inline plugins/ptp_operator/stats/stats.go:239]: should check state != nil

Comment (aneeshkp) [reply plugins/ptp_operator/stats/stats.go:239]: No required , ranging inputState (state) struct and iface is a pointer .

File: plugins/ptp_operator/metrics/metrics.go (hunk 1 above)
Comment (aneeshkp) [reply plugins/ptp_operator/metrics/metrics.go:134]: yes , it was debug log . Thanks will remove it

**Review by jzding — APPROVED** [review]
/lgtm

**Pull request merged**